Cem/single molecule

chris-erisyon opened the pull request:
Single molecule report code and notebooks

zack-erisyon commented on plaster/run/plots/single_molecule/single_molecule_report.py:
```diff
@@ -0,0 +1,813 @@
+#   - 'base_path' refers to the base path for the single_molecule_report directory
+#   - 'run_path' refers to the directory for the run
+
+JOBS_FOLDER = "/erisyon/internal/jobs_folder"
```
It definitely doesn't feel right do have any paths coded into these modules.
This and all of the following should go through interfaces.

zack-erisyon commented on plaster/run/plots/single_molecule/single_molecule_report.py:
```diff
@@ -0,0 +1,813 @@
+# ------------------------------
+
+
+def get_run_uuid(run_path):
```
This and the below should be coming from the a Job instance.

zack-erisyon commented on plaster/run/plots/single_molecule/single_molecule_report.py~:
```diff
@@ -0,0 +1,783 @@
+# Generate reports for individual traces in a run, organized into one HTML file per field/channel.
```
This needs to be removed. I assume it was added before the gitignore was setup

zack-erisyon reviewed (changes requested):
I'm not sure what our plan is. If this code is going to be main-lined then it seems like we need to put some work in to making it conform to existing patterns. For example, using APIs that exist, abstract out I/O, add tests.
If we think we're throwing this in for some quick gains and plan on deprecating it, that's a different story. In which case we should add a README that states our plan.
My guess is that this code is going to have utility and isn't going  anywhere for quite some time and if that's the case then we need to either spruce it up now or schedule time for that otherwise we run the risk of leaving these dangling issues to bite us before long, esp the path issues.

chris-erisyon commented:
Code updated as per our discussions:

- Lots of documentation added, including notes for future developers
- Moved the hardcoded paths to the local test code (in __main__)
- got UUID from job_manifest at the notebook level
- added _get_run_name() placeholder to get a canonical, human readable name for the runs. Will replace with final version once we have it.

-Chris

> On Mar 4, 2021, at 8:33 AM, zack-erisyon <notifications@github.com> wrote:
> 
> 
> @zack-erisyon requested changes on this pull request.
> 
> I'm not sure what our plan is. If this code is going to be main-lined then it seems like we need to put some work in to making it conform to existing patterns. For example, using APIs that exist, abstract out I/O, add tests.
> If we think we're throwing this in for some quick gains and plan on deprecating it, that's a different story. In which case we should add a README that states our plan.
> My guess is that this code is going to have utility and isn't going anywhere for quite some time and if that's the case then we need to either spruce it up now or schedule time for that otherwise we run the risk of leaving these dangling issues to bite us before long, esp the path issues.
> 
> In plaster/run/plots/single_molecule/single_molecule_report.py <https://github.com/erisyon/plaster/pull/229#discussion_r587510440>:
> 
> > +from plaster.tools.image import imops
> +from plaster.tools.image.coord import XY, HW
> +
> +from plaster.run.plots.single_molecule.single_molecule_report_template import (
> +    generate_single_molecule_report,
> +    generate_single_molecule_report_index,
> +    get_index_path,
> +)
> +
> +# Paths and names
> +
> +# In the functions below:
> +#   - 'base_path' refers to the base path for the single_molecule_report directory
> +#   - 'run_path' refers to the directory for the run
> +
> +JOBS_FOLDER = "/erisyon/internal/jobs_folder"
> It definitely doesn't feel right do have any paths coded into these modules.
> This and all of the following should go through interfaces.
> 
> In plaster/run/plots/single_molecule/single_molecule_report.py <https://github.com/erisyon/plaster/pull/229#discussion_r587511861>:
> 
> > +
> +# Color map for images
> +viridis = cm.get_cmap("viridis", 256)
> +viridis.set_under("w")  # anything under vmin should be white
> +
> +
> +# ------------------------------------------------------------
> +# Helpers
> +# ------------------------------------------------------------
> +
> +# ------------------------------
> +# Config/Index Files
> +# ------------------------------
> +
> +
> +def get_run_uuid(run_path):
> This and the below should be coming from the a Job instance.
> 
> In plaster/run/plots/single_molecule/single_molecule_report.py~ <https://github.com/erisyon/plaster/pull/229#discussion_r587514201>:
> 
> > @@ -0,0 +1,783 @@
> +# Generate reports for individual traces in a run, organized into one HTML file per field/channel.
> This needs to be removed. I assume it was added before the gitignore was setup
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/erisyon/plaster/pull/229#pullrequestreview-604153090>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APFW7SUDKBHEFO5GI5CLZXLTB6K4LANCNFSM4YSDJHHA>.
>

zack-erisyon reviewed (approved):
There's still a tilde file that should be removed

chris-erisyon commented:
Removed.

-Chris

> On Mar 4, 2021, at 11:49 AM, zack-erisyon <notifications@github.com> wrote:
> 
> 
> @zack-erisyon approved this pull request.
> 
> There's still a tilde file that should be removed
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/erisyon/plaster/pull/229#pullrequestreview-604383555>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APFW7SQ4EDBPZ7CWAGDLG33TB7BZDANCNFSM4YSDJHHA>.
>
